scene/resources/mesh_data_tool.cpp

```diff
@@ -289,6 +341,16 @@ Error MeshDataTool::commit_to_surface(const Ref<ArrayMesh> &p_mesh) {
 				bo[i * 4 + 2] = vtx.bones[2];
 				bo[i * 4 + 3] = vtx.bones[3];
 			}
+
+			for (uint32_t j = 0; j < Mesh::ARRAY_CUSTOM_COUNT; ++j) {
+				if (cuf_ptr[j] || cuu_ptr[j]) {
+					if (custom_types[j] > Mesh::ARRAY_CUSTOM_RGBA_HALF) {
+						memcpy(cuf_ptr[j] + i * custom_type_sizes[j], &vtx.custom[j].fd, custom_type_sizes[j] * sizeof(float));
+					} else {
+						memcpy(cuu_ptr[j] + i * custom_type_sizes[j], &vtx.custom[j].ud, custom_type_sizes[j]);
+					}
+				}
```

Comment:
nit: perhaps replace these if statements with simple null checks
```
				if (cuf_ptr[j]) {
					memcpy(cuf_ptr[j] + i * custom_type_sizes[j], &vtx.custom[j].fd, custom_type_sizes[j] * sizeof(float));
				} else if (cuu_ptr[j]) {
					memcpy(cuu_ptr[j] + i * custom_type_sizes[j], &vtx.custom[j].ud, custom_type_sizes[j]);
				}
```
Analogous to how lines 167-175 are done.